sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIO.java

```diff
@@ -654,94 +434,59 @@ public Write withExchange(String exchange, String exchangeType) {
      */
     public Write withExchange(String exchange) {
       checkArgument(exchange != null, "exchange can not be null");
-      return builder().setExchange(exchange).setExchangeDeclare(false).build();
-    }
-
-    /**
-     * Defines the queue where the messages will be sent. The queue has to be declared. It can be
-     * done by another application or by {@link RabbitMqIO} if you set {@link
-     * Write#withQueueDeclare} to {@code true}.
-     */
-    public Write withQueue(String queue) {
-      checkArgument(queue != null, "queue can not be null");
-      return builder().setQueue(queue).build();
-    }
-
-    /**
-     * If the queue is not declared by another application, {@link RabbitMqIO} can declare the queue
-     * itself.
-     *
-     * @param queueDeclare {@code true} to declare the queue, {@code false} else.
-     */
-    public Write withQueueDeclare(boolean queueDeclare) {
-      return builder().setQueueDeclare(queueDeclare).build();
+      return builder().setExchange(exchange).build();
     }
 
     @Override
     public PCollection<?> expand(PCollection<RabbitMqMessage> input) {
-      checkArgument(
-          exchange() != null || queue() != null, "Either exchange or queue has to be specified");
-      if (exchange() != null) {
-        checkArgument(queue() == null, "Queue can't be set in the same time as exchange");
-      }
-      if (queue() != null) {
-        checkArgument(exchange() == null, "Exchange can't be set in the same time as queue");
-      }
-      if (queueDeclare()) {
-        checkArgument(queue() != null, "Queue is required for the queue declare");
-      }
-      if (exchangeDeclare()) {
-        checkArgument(exchange() != null, "Exchange is required for the exchange declare");
-      }
       return input.apply(ParDo.of(new WriteFn(this)));
     }
 
     private static class WriteFn extends DoFn<RabbitMqMessage, Void> {
       private final Write spec;
 
-      private transient ConnectionHandler connectionHandler;
+      private UUID writerId;
+      private transient ChannelLeaser channelLeaser;
 
       WriteFn(Write spec) {
         this.spec = spec;
       }
 
       @Setup
       public void setup() throws Exception {
-        connectionHandler = new ConnectionHandler(spec.uri());
-        connectionHandler.start();
-
-        Channel channel = connectionHandler.getChannel();
-
-        if (spec.exchange() != null && spec.exchangeDeclare()) {
-          channel.exchangeDeclare(spec.exchange(), spec.exchangeType());
+        if (writerId == null) {
+          writerId = UUID.randomUUID();
         }
-        if (spec.queue() != null && spec.queueDeclare()) {
-          channel.queueDeclare(spec.queue(), true, false, false, null);
+        if (channelLeaser == null) {
+          channelLeaser = spec.connectionHandlerProviderFn().apply(null);
         }
+
+        channelLeaser.useChannel(writerId, ch -> null);
```

Comment:
is this here just so you can make sure a connection has been made?  might be worth a comment as it looks a little strange at first glance

Reply:
also tiny suggestion -- Java lacks `_` but I've mostly seen `ignored` used instead:
```suggestion
        channelLeaser.useChannel(writerId, ignored -> null);
```